bindings/kubernetes/kubernetes.go

```diff
@@ -66,22 +74,23 @@ func (k *kubernetesInput) Init(ctx context.Context, metadata bindings.Metadata)
 	return k.parseMetadata(metadata)
 }
 
-func (k *kubernetesInput) parseMetadata(metadata bindings.Metadata) error {
-	if val, ok := metadata.Properties["namespace"]; ok && val != "" {
-		k.namespace = val
-	} else {
-		return errors.New("namespace is missing in metadata")
-	}
-	if val, ok := metadata.Properties["resyncPeriodInSec"]; ok && val != "" {
-		intval, err := strconv.ParseInt(val, 10, 64)
-		if err != nil {
-			k.logger.Warnf("invalid resyncPeriodInSec %s; %v; defaulting to 10s", val, err)
-			k.resyncPeriod = time.Second * 10
+func (k *kubernetesInput) parseMetadata(meta bindings.Metadata) error {
+	m := kubernetesMetadata{}
+	err := metadata.DecodeMetadata(meta.Properties, &m)
+	if err != nil {
+		if strings.Contains(err.Error(), "decoding 'resyncPeriodInSec': time: invalid duration") {
```

Comment:
I would change this to:

```suggestion
		if strings.Contains(err.Error(), "resyncPeriodInSec") {
```

This is so if the upstream library (mapdecode) changes how it formats errors, this is less likely to break